.github/workflows/merge-schedule.yaml

```diff
@@ -11,13 +11,15 @@ on:
 jobs:
   merge_schedule:
     runs-on: ubuntu-latest
+    if:
+      ${{ github.event.pull_request.head.repo.full_name == github.repository }}
```

Comment:
Checks if the repo name of the pull request equals the github repo. I tested it with a fork (#3270) and [it appears to work correctly](https://github.com/iterative/dvc.org/actions/runs/1826967213)!